### PR TITLE
bug: VoidReturnFixer - fix for intervening attributes

### LINKED ISSUE #6690: PHP Attributes should not be treated as comments
## Bug report

The `@return` declared by phpdoc is not considered to be the return value of the function.

```shell
# ./vendor/bin/php-cs-fixer -V
PHP CS Fixer 3.13.0 Oliva by Fabien Potencier and Dariusz Ruminski.
PHP runtime: 8.0.23
```

**config:**

```php
<?php

declare(strict_types=1);

return (new PhpCsFixer\Config())
    ->setRules([
        '@PHP74Migration'            => true,
        '@PHP74Migration:risky'      => true,
        '@Symfony'                   => true,
        '@Symfony:risky'             => true,
        '@DoctrineAnnotation'        => true,
    ])
    ->setRiskyAllowed(true)
;
```

### Code snippet that reproduces the problem

**code:**

```php
<?php

declare(strict_types=1);

/**
 * @return mixed|false
 */
#[\ReturnTypeWillChange]
function test()
{
    throw new \RuntimeException('gg');
}
```

**result:**

```shell
# ./vendor/bin/php-cs-fixer fix --dry-run --diff test.php 
Loaded config default from "/mnt/d/projects/test/.php-cs-fixer.php".
Using cache file ".php-cs-fixer.cache".
   1) test.php
      ---------- begin diff ----------
--- /mnt/d/projects/test/test.php
+++ /mnt/d/projects/test/test.php
@@ -6,7 +6,7 @@
  * @return mixed|false
  */
 #[\ReturnTypeWillChange]
-function test()
+function test(): void
 {
     throw new \RuntimeException('gg');
 }

      ----------- end diff -----------


Checked all files in 0.021 seconds, 14.000 MB memory used
```

### PR DESCRIPTION
**Pull request by paulbalandan**
Fixes #6690

### CODE REVIEW

**Review by kubawerlos — APPROVED** [review]

**Review by Wirone — CHANGES_REQUESTED** [review]
I believe it would be good if there was a integration test that would ensure that `@return void` is removed when `no_superfluous_phpdoc_tags` fixer is enabled 🤔. WDYT @kubawerlos?

File: src/Fixer/FunctionNotation/VoidReturnFixer.php [hunk 1]
```diff
@@ -236,16 +236,26 @@ private function fixFunctionDefinition(Tokens $tokens, int $index): void
      */
     private function findReturnAnnotations(Tokens $tokens, int $index): array
     {
-        do {
-            $index = $tokens->getPrevNonWhitespace($index);
-        } while ($tokens[$index]->isGivenKind([
+        static $previousTokens = [
```
Comment (mvorisek) [inline src/Fixer/FunctionNotation/VoidReturnFixer.php:239]: is `static` really faster/needed?

Comment (paulbalandan) [reply src/Fixer/FunctionNotation/VoidReturnFixer.php]: Thanks!

**Review by Wirone — APPROVED** [review]

**Comment by Wirone** [comment]
Thank you, @paulbalandan 🍻

**Pull request merged**